src/Tests/Microsoft.DotNet.ApiCompatibility.Tests/Rules/CannotRemoveBaseTypeOrInterfaceTests.cs

```diff
@@ -0,0 +1,294 @@
+  public interface IFirstInterface { }
+  public interface INewInterface : IFirstInterface { }
+}
+";
+
+            IAssemblySymbol left = SymbolFactory.GetAssemblyFromSyntax(leftSyntax);
+            IAssemblySymbol right = SymbolFactory.GetAssemblyFromSyntax(rightSyntax);
+
+            ApiComparer differ = new();
+
+            Assert.Empty(differ.GetDifferences(left, right));
+        }
+
+        [Fact]
+        public void RemovedInterfaceAndBaseClassAreReported()
+        {
+            string leftSyntax = @"
+namespace CompatTests
+{
+  public class First : FirstBase, IFirstInterface { }
+  public class FirstBase { }
+  public interface IFirstInterface { }
+}
+";
+
+            string rightSyntax = @"
+namespace CompatTests
+{
+  public class First : NewBase { }
+  public class NewBase : INewInterface { }
+  public class FirstBase { }
+  public interface IFirstInterface { }
+  public interface INewInterface { }
+}
+";
+
+            IAssemblySymbol left = SymbolFactory.GetAssemblyFromSyntax(leftSyntax);
+            IAssemblySymbol right = SymbolFactory.GetAssemblyFromSyntax(rightSyntax);
+
+            ApiComparer differ = new();
+
+            CompatDifference[] differences = differ.GetDifferences(left, right).ToArray();
+
+            CompatDifference[] expected = new[]
+            {
+                new CompatDifference(DiagnosticIds.CannotRemoveBaseType, string.Empty, DifferenceType.Changed, "T:CompatTests.First"),
+                new CompatDifference(DiagnosticIds.CannotRemoveBaseInterface, string.Empty, DifferenceType.Changed, "T:CompatTests.First"),
+            };
+
+            Assert.Equal(expected, differences, CompatDifferenceComparer.Default);
+
+            differences[0].Message.Contains("CompatTests.FirstBase");
+            differences[1].Message.Contains("CompatTests.IFirstInterface");
+        }
+
+        [Fact]
+        public void RemovedInternalInterfaceIsNotReported()
+        {
+            string leftSyntax = @"
+namespace CompatTests
+{
+  public class First : IFirstInternalInterface, IFirstInterface { }
+  public interface IFirstInterface { }
+  internal interface IFirstInternalInterface { }
+}
+";
+
+            string rightSyntax = @"
+namespace CompatTests
+{
+  public class First : IFirstInterface { }
+  public interface IFirstInterface { }
+}
+";
+
+            IAssemblySymbol left = SymbolFactory.GetAssemblyFromSyntax(leftSyntax);
+            IAssemblySymbol right = SymbolFactory.GetAssemblyFromSyntax(rightSyntax);
+
+            ApiComparer differ = new();
+
+            Assert.Empty(differ.GetDifferences(left, right));
+        }
+
+        [Fact]
+        public void RemovedFromLeftReportedOnStrictMode()
+        {
+            string leftSyntax = @"
+namespace CompatTests
+{
+  public class First { }
+  public class FirstBase { }
+  public interface IFirstInterface { }
+}
+";
+
+            string rightSyntax = @"
+namespace CompatTests
+{
+  public class First : FirstBase, IFirstInterface { }
```

Comment:
are you worried that this wont allow developers to derive from new classes while running in the strict mode, should we allow disabling the strict mode for particular rules.

Reply:
> are you worried that this wont allow developers to derive from new classes while running in the strict mode, should we allow disabling the strict mode for particular rules.

It would allow them to derive from new classes, but they should derive from those new classes on both sides. `StrictMode` is intended for a compile vs runtime asset scenario.